### ISSUE TITLE
TileCache offmesh conenctions.

### ISSUE THREAD

**Issue by mitm001** [ppiastucki/recast4j#59]
Hi Piotr,

Starting to learn the tile cache stuff and while reading I see this,
[TileCacheMeshProcess.java#L25](https://github.com/ppiastucki/recast4j/blob/d7c652d82e58f4fa768c395efeb13a0877cd0704/detourtilecache/src/main/java/org/recast4j/detour/tilecache/TileCacheMeshProcess.java#L25)

but recast has this where we set the poly flags and offmesh connections.
[Sample_TempObstacles.cpp#L192](https://github.com/recastnavigation/recastnavigation/blob/56dbf4c482741f06d10528f86e953ac57f86a3cb/RecastDemo/Source/Sample_TempObstacles.cpp#L192)

when reading NavMesh addTile() I see how connect the links for offmesh connections,
[NavMesh.java#L478](https://github.com/ppiastucki/recast4j/blob/3c532068d79fe0306fedf035e50216008c306cdf/detour/src/main/java/org/recast4j/detour/NavMesh.java#L478)

but in TileCache addTile(), I dont see how you are fixing up the links?
[TileCache.java#L208](https://github.com/ppiastucki/recast4j/blob/d7c652d82e58f4fa768c395efeb13a0877cd0704/detourtilecache/src/main/java/org/recast4j/detour/tilecache/TileCache.java#L208)

I seem to be missing how this works.

What is the process here for setting area types, flags, and offmesh connections with Recast4j tile cache?

**Comment by ppiastucki** [comment]
TileCacheMeshProcess is just an interface, you can pass whatever implementation you like. It may set flags and areas to whatever you want. There is no need to pass flags and areas as separate arguments as they are already available as fields in NavMeshDataCreateParams class, even in C++:
See: https://github.com/recastnavigation/recastnavigation/blob/05b2b8da80037887d3e79af8a59a8f0f8ed02602/DetourTileCache/Source/DetourTileCache.cpp#L735

NavMesh::addTile is a totally different method than TileCache::addTile.TileCache hold precomputed heightfields and to turn them into nav mesh tiles TileCache::buildNavMeshTile is invoked, see https://github.com/ppiastucki/recast4j/blob/d7c652d82e58f4fa768c395efeb13a0877cd0704/detourtilecache/src/main/java/org/recast4j/detour/tilecache/TileCache.java#L480.

I hope the info is helpful.

**Comment by mitm001** [comment]
> as they are already available as fields in NavMeshDataCreateParams

Yes, Area types are set when buildSolidHeightfield is called from TileLayerBuilders getHeightfieldSet. 

I had to make some changes to those since my navmesh has multiple area types and your current implementation expects a single mesh with a single area modification. I changed them to accept triangle lengths paired with an area modification so I can mark triangles in buildSolidHeightfield. 

This was something Mikko said he intended to do eventually but then he lost interest in his project before implementing it.

I am almost done implementing the offmesh connections for tile cache.  This is a little more difficult to do since I call [TestTileLayerBuilder.java#L64](https://github.com/ppiastucki/recast4j/blob/master/detourtilecache/src/test/java/org/recast4j/detour/tilecache/TestTileLayerBuilder.java#L64) to build the mesh rather than the AbstractTileLayersBuilder build.
